mne/preprocessing/tests/test_stim.py

```diff
@@ -25,52 +27,105 @@ def test_stim_elim():
-    tmax = 0.015
+    tmin, tmax= -0.045, 0.015
     test_tminidx = int(-0.035 * raw.info['sfreq'])
     test_tmaxidx = int(-0.015 * raw.info['sfreq'])
 
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='linear')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     diff_data0 = np.diff(data[0])
     diff_data0 -= np.mean(diff_data0)
     assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='window')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     assert_true(np.all(data) == 0.)
 
     # use window after stimulus
-    tmin = 0.005
-    tmax = 0.045
+    tmin, tmax = 0.005, 0.045
     test_tminidx = int(0.015 * raw.info['sfreq'])
     test_tmaxidx = int(0.035 * raw.info['sfreq'])
 
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='linear')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     diff_data0 = np.diff(data[0])
     diff_data0 -= np.mean(diff_data0)
     assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='window')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     assert_true(np.all(data) == 0.)
+
+
+def test_stim_fix_evoked():
+    """Test eliminate stim artifact"""
+    raw = Raw(raw_fname, preload=True)
+    events = read_events(event_fname)
+    tmin, tmax, event_id = -0.2, 0.5, 1
+    picks = pick_types(raw.info, meg=True, eeg=True,
+                       eog=True, stim=False, exclude='bads')
+    epochs = Epochs(raw, events, event_id, tmin, tmax,
+                    picks=picks, preload=False)
+
+    # use window before stimulus
+    evoked = epochs.average()
+    tmin, tmax= -0.2, 0.1
+    test_tminidx = int(-0.1 * evoked.info['sfreq']) - evoked.first
+    test_tmaxidx = int(0.05 * evoked.info['sfreq']) - evoked.first
+    evoked = fix_stim_artifact(evoked, tmin, tmax, mode='window')
+    data = evoked.data[:, test_tminidx:test_tmaxidx]
+    assert_true(np.all(data) == 0.)
+
+    # use window after stimulus
+    tmin, tmax= 0.2, 0.4
+    test_tminidx = int(0.25 * evoked.info['sfreq']) - evoked.first
+    test_tmaxidx = int(0.35 * evoked.info['sfreq']) - evoked.first
+    evoked = fix_stim_artifact(evoked, tmin, tmax, mode='window')
+    data = evoked.data[:, test_tminidx:test_tmaxidx]
+    assert_true(np.all(data) == 0.)
+
+
+def test_stim_fix_epochs():
```

Comment:
let's just have one function to test all three. That way we don't replicate the same steps in the pipeline and save testing time + keep the code concise.